### DISCUSSION TITLE
How to add Dokka plugins

### DISCUSSION THREAD

**Discussion by odenix** [adamko-dev/dokkatoo#175] [Q&A] [answered]
How do I add Dokka plugins with Dokkatoo? Specifically I'd like to use the kotlin-as-java Dokka plugin.

I'm confused as to why every example under dokkatoo/examples has both a dokka/ and a dokkatoo/ directory, and why the dokkatoo/ directory of the kotlin-as-java example is almost empty. I tried:

```
dependencies {
  dokkaPlugin("org.jetbrains.dokka:kotlin-as-java-plugin:1.9.10")
}
```
But Gradle says it cannot resolve `dokkaPlugin`. Same result for `dokkatooPlugin`.

**Comment by odenix** [comment]
Answering my own question: `dokkatooPluginHtml` seems to work.

**Reply by aSemy** [reply, marked as answer]
Hi 👋, thanks for the questions!

> How do I add Dokka plugins with Dokkatoo? Specifically I'd like to use the kotlin-as-java Dokka plugin.

Yeah... it's not very clear. It's been a while since I looked at it, and I had to do some digging.

As you discovered, Dokka Plugins are only applicable to each publication formats. There's four built-in formats (HTML, GFM, Jekyll, Javadoc). So you'll have to do the following:

1. apply the format specific Dokkatoo plugin (e.g. `id("dev.adamko.dokkatoo-html")`), or the 'all formats' plugin `id("dev.adamko.dokkatoo")`,
2. then add the plugin to the format specific dependency containers, e.g. `dokkatooPluginHtml("org.jetbrains.dokka:kotlin-as-java-plugin:1.9.10")`.

I could add a general `dokkatooPlugin()` dependency for plugins that need to be applied to all formats. Please [make a new issue](https://github.com/adamko-dev/dokkatoo/issues/new) if such a feature would help.

> I'm confused as to why every example under dokkatoo/examples has both a dokka/ and a dokkatoo/ directory, and why the dokkatoo/ directory of the kotlin-as-java example is almost empty. I tried:

Indeed, the docs are lacking https://github.com/adamko-dev/dokkatoo/issues/121

To quickly explain: The examples are based on [the existing Dokka examples](https://github.com/Kotlin/dokka/tree/master/examples). I split each of those examples into a Dokka and Dokkatoo version for two reasons. First, I wanted to verify that Dokkatoo and Dokka both generate the same output. Second, I wanted to make it easier for people to migrate from Dokka to Dokkatoo, so they can see the equivalent Dokkatoo code.

However, I haven't converted all the examples, which is why some of the Dokkatoo directories are almost empty.

**Reply by odenix** [reply]
Thanks for the detailed answer! After spending a day with Dokkatoo, I can confidently say that what would help me most is external docs for all config options. Editing `.gradle.kts` files (esp. those in `buildSrc`) in IntelliJ is a nightmare: dead slow, syntax highlighting/code navigation/code completion comes and goes, cryptic compile errors, internal IntelliJ errors. Hence external docs would be a life saver.

**Reply by aSemy** [reply]
Hey 👋 

I've added a docs site with some guides and docs! It's not complete, but there's a lot of info there, as well as a showcase so you can view not only Dokkatoo's examples, but also real-life projects.

https://adamko-dev.github.io/dokkatoo/

If you think something is missing, then please make an issue to describe what you want to see, or even better, a PR! Contributions would be really valued ❤️

**Reply by odenix** [reply]
Wow, looks amazing!!!
